pandas/_libs/src/ujson/lib/ultrajsondec.c

```diff
@@ -64,6 +65,7 @@ struct DecoderState {
     int escHeap;
     int lastType;
     JSUINT32 objDepth;
+    char *cStr; // storage for BigNum
```

Comment:
No need to change the struct here; just look at the existing implementation of `decode_string` and mirror that